packages/flutter/lib/src/material/chip.dart

```diff
@@ -1892,22 +1892,33 @@ class _RenderChip extends RenderBox {
   }
 
   Size _layoutLabel(double iconSizes, Size size) {
+    final Size rawSize = _boxSize(label);
     // Now that we know the label height and the width of the icons, we can
     // determine how much to shrink the width constraints for the "real" layout.
     if (constraints.maxWidth.isFinite) {
       label.layout(
-        constraints.loosen().copyWith(
-            maxWidth: math.max(
-              0.0,
-              constraints.maxWidth - iconSizes - theme.labelPadding.horizontal,
-            ),
-            maxHeight: size.height),
+        constraints.copyWith(
```

Comment:
You don't want copyWith here, since you are literally overriding every value. :-) Just call the constructor.